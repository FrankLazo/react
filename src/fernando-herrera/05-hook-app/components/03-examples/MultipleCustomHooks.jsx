import useCounter from '../../hooks/useCounter';
import useFetch from '../../hooks/useFetch';

const MultipleCustomHooks = () =>
{
	const url = 'https://www.breakingbadapi.com/api/';
	const { state, increment } = useCounter(1);
	const { loading, data } = useFetch(`${ url }quotes/${ state }`);
	const { author, quote } = !!data && data[0];

	return (
		<div className="min-height-full content has-background-dark is-flex is-flex-direction-column is-align-items-start p-6 has-text-light">
			<h1 className="has-text-light">Breaking Bad Quotes</h1>
			<hr className="full-rule has-background-grey-dark" />

		{
			loading
			?
			<div className="notification is-warning is-light">
				Loading...
			</div>
			:
			<blockquote className="has-text-dark">
				<p>{ quote }</p>
				<p className="has-text-weight-bold">{ author }</p>
			</blockquote>
		}

			<button
				className="button is-primary"
				onClick={ () => increment() }
				>
				Next quote
			</button>
		</div>
	);
};

export default MultipleCustomHooks;
