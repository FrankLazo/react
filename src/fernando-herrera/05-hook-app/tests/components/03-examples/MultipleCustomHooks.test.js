import { shallow } from 'enzyme';
import MultipleCustomHooks from '../../../components/03-examples/MultipleCustomHooks';
import useCounter from '../../../hooks/useCounter';
import useFetch from '../../../hooks/useFetch';

jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');

describe('<MultipleCustomHooks /> tests', () => {
	beforeEach(() => {
		useCounter.mockReturnValue({
			state: 10,
			increment: () => {},
			decrement: () => {},
			reset: () => {},
		});
	});

	test('should show the right component', () => {
		useFetch.mockReturnValue({
			data: null,
			loading: true,
			error: null,
		});

		const wrapper = shallow(<MultipleCustomHooks />);
		expect(wrapper).toMatchSnapshot();
	});

	test('should show the information', () => {
		useFetch.mockReturnValue({
			data: [{
				author: 'Frank Lazo',
				quote: 'Hello world!',
			}],
			loading: false,
			error: null,
		});

		const wrapper = shallow(<MultipleCustomHooks />);

		expect(wrapper.find('.notification').exists()).toBe(false);
		expect(wrapper.find('blockquote p:first-child').text().trim()).toBe('Hello world!');
		expect(wrapper.find('blockquote p:last-child').text().trim()).toBe('Frank Lazo');
	});

});
