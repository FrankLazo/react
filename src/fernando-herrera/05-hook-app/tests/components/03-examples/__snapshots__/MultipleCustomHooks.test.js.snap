// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultipleCustomHooks /> tests should show the right component 1`] = `
<div
  className="min-height-full content has-background-dark is-flex is-flex-direction-column is-align-items-start p-6 has-text-light"
>
  <h1
    className="has-text-light"
  >
    Breaking Bad Quotes
  </h1>
  <hr
    className="full-rule has-background-grey-dark"
  />
  <div
    className="notification is-warning is-light"
  >
    Loading...
  </div>
  <button
    className="button is-primary"
    onClick={[Function]}
  >
    Next quote
  </button>
</div>
`;
